pkg/cmd/server/origin/master.go

```diff
@@ -411,35 +411,60 @@ func (c *MasterConfig) ensureMasterAuthorizationNamespace() {
 // ensureComponentAuthorizationRules initializes the global policies
 func (c *MasterConfig) ensureComponentAuthorizationRules() {
 	registry := authorizationetcd.New(c.EtcdHelper)
-	ctx := kapi.WithNamespace(kapi.NewContext(), c.MasterAuthorizationNamespace)
 
-	if existing, err := registry.GetPolicy(ctx, authorizationapi.PolicyName); err == nil || strings.Contains(err.Error(), " not found") {
-		if existing != nil && existing.Name == authorizationapi.PolicyName {
-			return
-		}
+	roleRegistry := roleregistry.NewVirtualRegistry(registry)
+	for _, role := range bootstrappolicy.GetBootstrapRoles(c.MasterAuthorizationNamespace, c.OpenshiftSharedResourcesNamespace) {
+		ctx := kapi.WithNamespace(kapi.NewContext(), role.Namespace)
 
-		bootstrapGlobalPolicy := authorizer.GetBootstrapPolicy(c.MasterAuthorizationNamespace)
-		if err = registry.CreatePolicy(ctx, bootstrapGlobalPolicy); err != nil {
-			glog.Errorf("Error creating policy: %v due to %v\n", bootstrapGlobalPolicy, err)
-		}
+		if _, err := roleRegistry.GetRole(ctx, role.Name); kapierror.IsNotFound(err) {
+			if err := roleRegistry.CreateRole(ctx, &role); err != nil {
+				glog.Errorf("Error creating role: %#v due to %v\n", role, err)
+			}
 
-	} else {
-		glog.Errorf("Error getting policy: %v due to %v\n", authorizationapi.PolicyName, err)
+		} else if err != nil {
+			glog.Errorf("Error get role: %#v due to %v\n", role, err)
+		}
 	}
 
-	if existing, err := registry.GetPolicyBinding(ctx, c.MasterAuthorizationNamespace); err == nil || strings.Contains(err.Error(), " not found") {
-		if existing != nil && existing.Name == c.MasterAuthorizationNamespace {
-			return
-		}
+	roleBindingRegistry := rolebindingregistry.NewVirtualRegistry(registry, registry, c.MasterAuthorizationNamespace)
+	for _, roleBinding := range bootstrappolicy.GetBootstrapRoleBindings(c.MasterAuthorizationNamespace, c.OpenshiftSharedResourcesNamespace) {
+		ctx := kapi.WithNamespace(kapi.NewContext(), roleBinding.Namespace)
 
-		bootstrapGlobalPolicyBinding := authorizer.GetBootstrapPolicyBinding(c.MasterAuthorizationNamespace)
-		if err = registry.CreatePolicyBinding(ctx, bootstrapGlobalPolicyBinding); err != nil {
-			glog.Errorf("Error creating policy: %v due to %v\n", bootstrapGlobalPolicyBinding, err)
-		}
+		if _, err := roleBindingRegistry.GetRoleBinding(ctx, roleBinding.Name); kapierror.IsNotFound(err) {
 
-	} else {
-		glog.Errorf("Error getting policy: %v due to %v\n", c.MasterAuthorizationNamespace, err)
+			// if this is a binding for a non-master namespace the policy binding must be provisioned
+			if roleBinding.Namespace != c.MasterAuthorizationNamespace {
```

Comment:
Only for the bindings made against master namespaced roles.  Otherwise, it must exist ahead of time.  That allows behavior where a project admin can create any roleBinding he wants, but only against master roles.